src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -0,0 +1,94 @@
+#include "generate_pruning_statistics.hpp"
+
+#include <atomic>
+#include <iostream>
+#include <thread>
+#include <unordered_set>
+
+#include "operators/table_wrapper.hpp"
+#include "resolve_type.hpp"
+#include "statistics/attribute_statistics.hpp"
+#include "statistics/statistics_objects/equal_distinct_count_histogram.hpp"
+#include "statistics/statistics_objects/generic_histogram_builder.hpp"
+#include "statistics/statistics_objects/min_max_filter.hpp"
+#include "statistics/statistics_objects/null_value_ratio_statistics.hpp"
+#include "statistics/statistics_objects/range_filter.hpp"
+#include "statistics/table_statistics.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/table.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+template <typename T>
+void create_pruning_filter_for_segment(AttributeStatistics<T>& segment_statistics, const pmr_vector<T>& dictionary) {
+  std::shared_ptr<AbstractStatisticsObject> pruning_filter;
+  if constexpr (std::is_arithmetic_v<T>) {
+    pruning_filter = RangeFilter<T>::build_filter(dictionary);
+  } else {
+    if (!dictionary.empty()) {
+      pruning_filter = std::make_shared<MinMaxFilter<T>>(dictionary.front(), dictionary.back());
+    }
+  }
+
+  if (pruning_filter) {
+    segment_statistics.set_statistics_object(pruning_filter);
+  }
+}
+
+}  // namespace
+
+namespace opossum {
+
+void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
```

Comment:
To avoid confusion, all variable names now use the term "pruning statistics". I am not really comfortable with "Filter" meaning "statistics object usable for pruning", but that's beyond this PR.